nixos/modules/services/web-apps/discourse.nix

```diff
@@ -0,0 +1,120 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.discourse;
+
+  bundle = "${cfg.package}/share/discourse/bin/bundle";
+
+in
+
+{
+  options = {
+    services.discourse = {
+      enable = mkOption {
+        type = types.bool;
+        default = false;
+        description = "Enable the Discourse service.";
+      };
+
+      package = mkOption {
+        type = types.package;
+        default = pkgs.discourse;
+        defaultText = "pkgs.discourse";
+        description = ''
+          Which Discourse package to use.
+        '';
+        example = "pkgs.discourse.override { ruby = pkgs.ruby_2_5; }";
+      };
+
+      port = mkOption {
+        type = types.int;
+        default = 3000;
+        description = "Port on which Discourse is ran.";
+      };
+
+      stateDir = mkOption {
+        type = types.str;
+        default = "/var/lib/discourse";
+        description = "The state directory, logs and plugins are stored here.";
```

Comment:
Unless there's a good reason this should be configurable, I'd rather have it not be. Then we can use systemd's `StateDirectory` to manage permissions and such.